File: src/main/java/quebec/virtualite/backend/Application.java
package quebec.virtualite.backend;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import quebec.virtualite.backend.security.SecurityUserManager;

import javax.annotation.PostConstruct;

import static quebec.virtualite.backend.security.SecurityUsers.TEST_PASSWORD;
import static quebec.virtualite.backend.security.SecurityUsers.TEST_USER;

@SpringBootApplication(scanBasePackages = {"quebec.virtualite.*"})
@Slf4j
public class Application
{
    @Autowired
    private SecurityUserManager userManager;

    static
    {
        log.warn("STARTING...");
    }

    public static void main(final String[] args)
    {
        SpringApplication.run(Application.class, args);
    }

    @PostConstruct
    public void init()
    {
        userManager.defineUser(TEST_USER, TEST_PASSWORD);

        log.warn("STARTED");
    }
}

File: src/main/java/quebec/virtualite/backend/Application.kt
package quebec.virtualite.backend

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import quebec.virtualite.backend.Application
import quebec.virtualite.backend.security.SecurityUserManager
import quebec.virtualite.backend.security.SecurityUsers.TEST_PASSWORD
import quebec.virtualite.backend.security.SecurityUsers.TEST_USER

@SpringBootApplication(scanBasePackages = ["quebec.virtualite.*"])
open class Application(userManager: SecurityUserManager)
{
    init
    {
        userManager.defineUser(TEST_USER, TEST_PASSWORD)

        // FIXME-0 Implement logging
//        log.warn("STARTED")
    }

    companion object
    {
        init
        {
//            Application.log.warn("STARTING...")
        }

        @JvmStatic
        fun main(args: Array<String>)
        {
            SpringApplication.run(Application::class.java, *args)
        }
    }
}
